Firmware/ControladorRobo/irarray.cpp

```diff
@@ -79,15 +79,28 @@ void IRArray::readSensors()
     }
 }
 
-uint16_t* IRArray::sensor(uint8_t i)
+uint16_t IRArray::sensor(uint8_t i, uint8_t typ = 0)
 {
     if( i < sizeof(m_sensor))
     {
-        return &m_sensor[i];
+        switch(typ)
+        {
+            case NORMAL:
+                return m_sensor[i];
+            break;
+
+            case MAX:
+                return m_sensorHigh[i];
+            break;
+
+            case MIN:
+                return m_sensorLow[i];
+            break;
+        }
     }
     else
     {
-        return nullptr;
+        return 0;
     }
```

Comment:
If i < sizeof(m_sensor) and typ != NORMAL, MAX, MIN. It will return something crazy, as a default behaviour I think that it's better to return NORMAL.
```
case NORMAL:
default:
     return m_sensor[i]
break;
```